runtime-modules/storage/src/tests/mod.rs

```diff
@@ -6145,3 +6148,111 @@ fn funds_needed_for_upload_succeeds_overflow_sat() {
         assert_eq!(funds_needed_saturated, u64::MAX);
     });
 }
+
+#[test]
+fn new_obj_deletion_prize_have_no_effect_on_older_data_obj() {
```

Comment:
I feel that this test should be reworked:
0. first off use something like `delete_data_object_succeeds_with_original_obj_deletion_prize` as test name
1. Setup your scenario with one dynamic bag containing some > 1 (in order to make sure that all deletion prize loops used in calculation works as expected) objects @ a given deletion prize (Arrange part)
2. change deletion prize (Arrange part)
3. call `delete_data_object` (Act part)
4. assert variation in balance is proportional to the original deletion prize (Assert part)

Reply:
This test has been refactored, it has only one bag now with 5 objects.
deletion prize changes,
3 objects  are removed,
in the end we have only one assert.